From 193fe11bbd2587d4f75e6446f5f582df1641e3ca Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Mon, 4 Feb 2019 18:57:21 +0530
Subject: [PATCH 0948/1051] octeontx2: Sync directory structure of octeontx2 AF
 and PF drivers to upstream

- Updated location
  AF driver: drivers/net/ethernet/marvell/octeontx2/af
  PF/VF driver: drivers/net/ethernet/marvell/octeontx2/nic

- Few misc changes in Kconfig, Makefile

Change-Id: I713a6e469910706078c8a018cb6181460b3bde13
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3544
Reviewed-by: Linu Cherian <Linu.Cherian@cavium.com>
Tested-by: Linu Cherian <Linu.Cherian@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/cavium/cpt/9x/Makefile         |  4 +--
 drivers/net/ethernet/marvell/Kconfig          | 12 +-------
 drivers/net/ethernet/marvell/Makefile         |  2 +-
 .../net/ethernet/marvell/octeontx2/Kconfig    | 29 +++++++++++++++++++
 .../net/ethernet/marvell/octeontx2/Makefile   | 12 ++------
 .../ethernet/marvell/octeontx2/af}/Makefile   |  0
 .../ethernet/marvell/octeontx2/af}/cgx.c      |  0
 .../ethernet/marvell/octeontx2/af}/cgx.h      |  0
 .../marvell/octeontx2/af}/cgx_fw_if.h         |  0
 .../ethernet/marvell/octeontx2/af}/common.h   |  0
 .../ethernet/marvell/octeontx2/af}/mbox.c     |  0
 .../ethernet/marvell/octeontx2/af}/mbox.h     |  0
 .../ethernet/marvell/octeontx2/af}/npc.h      |  0
 .../marvell/octeontx2/af}/npc_profile.h       |  0
 .../ethernet/marvell/octeontx2/af}/ptp.c      |  0
 .../ethernet/marvell/octeontx2/af}/ptp.h      |  0
 .../ethernet/marvell/octeontx2/af}/rvu.c      |  0
 .../ethernet/marvell/octeontx2/af}/rvu.h      |  0
 .../ethernet/marvell/octeontx2/af}/rvu_cgx.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_cpt.c  |  0
 .../marvell/octeontx2/af}/rvu_debugfs.c       |  0
 .../ethernet/marvell/octeontx2/af}/rvu_ndc.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_nix.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_npa.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_npc.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_ptp.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_reg.c  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_reg.h  |  0
 .../ethernet/marvell/octeontx2/af}/rvu_sso.c  |  0
 .../marvell/octeontx2/af}/rvu_struct.h        |  0
 .../ethernet/marvell/octeontx2/af}/rvu_tim.c  |  0
 .../marvell/octeontx2/af}/rvu_validation.c    |  0
 .../marvell/octeontx2/af}/rvu_validation.h    |  0
 .../ethernet/marvell/octeontx2/nic/Makefile   | 13 +++++++++
 .../marvell/octeontx2/{ => nic}/otx2_common.c |  0
 .../marvell/octeontx2/{ => nic}/otx2_common.h |  0
 .../octeontx2/{ => nic}/otx2_ethtool.c        |  0
 .../marvell/octeontx2/{ => nic}/otx2_pf.c     |  0
 .../marvell/octeontx2/{ => nic}/otx2_ptp.c    |  0
 .../marvell/octeontx2/{ => nic}/otx2_ptp.h    |  0
 .../marvell/octeontx2/{ => nic}/otx2_reg.h    |  0
 .../marvell/octeontx2/{ => nic}/otx2_struct.h |  0
 .../marvell/octeontx2/{ => nic}/otx2_txrx.c   |  0
 .../marvell/octeontx2/{ => nic}/otx2_txrx.h   |  0
 .../marvell/octeontx2/{ => nic}/otx2_vf.c     |  0
 drivers/soc/marvell/Kconfig                   | 11 -------
 drivers/soc/marvell/Makefile                  |  1 -
 drivers/soc/marvell/octeontx2-rm/Makefile     |  2 +-
 48 files changed, 50 insertions(+), 36 deletions(-)
 create mode 100644 drivers/net/ethernet/marvell/octeontx2/Kconfig
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/Makefile (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/cgx.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/cgx.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/cgx_fw_if.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/common.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/mbox.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/mbox.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/npc.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/npc_profile.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/ptp.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/ptp.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_cgx.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_cpt.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_debugfs.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_ndc.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_nix.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_npa.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_npc.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_ptp.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_reg.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_reg.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_sso.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_struct.h (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_tim.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_validation.c (100%)
 rename drivers/{soc/marvell/octeontx2 => net/ethernet/marvell/octeontx2/af}/rvu_validation.h (100%)
 create mode 100644 drivers/net/ethernet/marvell/octeontx2/nic/Makefile
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_common.c (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_common.h (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_ethtool.c (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_pf.c (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_ptp.c (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_ptp.h (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_reg.h (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_struct.h (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_txrx.c (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_txrx.h (100%)
 rename drivers/net/ethernet/marvell/octeontx2/{ => nic}/otx2_vf.c (100%)

diff --git a/drivers/crypto/cavium/cpt/9x/Makefile b/drivers/crypto/cavium/cpt/9x/Makefile
index 720725735e4e..b623b98709a6 100644
--- a/drivers/crypto/cavium/cpt/9x/Makefile
+++ b/drivers/crypto/cavium/cpt/9x/Makefile
@@ -10,5 +10,5 @@ ifeq ($(CONFIG_CRYPTO_DEV_OCTEONTX2_CPT), m)
 endif
 
 ccflags-y := -I$(src)/../common/
-ccflags-y += -I$(src)/../../../../soc/marvell/octeontx2/
-ccflags-y += -I$(src)/../../../../net/ethernet/marvell/octeontx2/
+ccflags-y += -I$(src)/../../../../net/ethernet/marvell/octeontx2/af/
+ccflags-y += -I$(src)/../../../../net/ethernet/marvell/octeontx2/nic/
diff --git a/drivers/net/ethernet/marvell/Kconfig b/drivers/net/ethernet/marvell/Kconfig
index 96fc1cab3f51..3238aa7f5dac 100644
--- a/drivers/net/ethernet/marvell/Kconfig
+++ b/drivers/net/ethernet/marvell/Kconfig
@@ -167,17 +167,7 @@ config SKY2_DEBUG
 
 	  If unsure, say N.
 
-config OCTEONTX2_PF
-	tristate "Marvell OcteonTX2 NIC physical function driver"
-	select OCTEONTX2_MBOX
-	depends on PCI && ARM64 && ARM64_LSE_ATOMICS
-	---help---
-	  This driver supports Marvell's OcteonTX2 NIC physical function.
 
-config OCTEONTX2_VF
-	tristate "Marvell OcteonTX2 NIC virtual function driver"
-	depends on OCTEONTX2_PF
-	---help---
-	  This driver supports Marvell's OcteonTX2 NIC virtual function.
+source "drivers/net/ethernet/marvell/octeontx2/Kconfig"
 
 endif # NET_VENDOR_MARVELL
diff --git a/drivers/net/ethernet/marvell/Makefile b/drivers/net/ethernet/marvell/Makefile
index afe726dea864..89dea7284d5b 100644
--- a/drivers/net/ethernet/marvell/Makefile
+++ b/drivers/net/ethernet/marvell/Makefile
@@ -11,4 +11,4 @@ obj-$(CONFIG_MVPP2) += mvpp2/
 obj-$(CONFIG_PXA168_ETH) += pxa168_eth.o
 obj-$(CONFIG_SKGE) += skge.o
 obj-$(CONFIG_SKY2) += sky2.o
-obj-$(CONFIG_NET_VENDOR_MARVELL) += octeontx2/
+obj-y		+= octeontx2/
diff --git a/drivers/net/ethernet/marvell/octeontx2/Kconfig b/drivers/net/ethernet/marvell/octeontx2/Kconfig
new file mode 100644
index 000000000000..106f6b39c62f
--- /dev/null
+++ b/drivers/net/ethernet/marvell/octeontx2/Kconfig
@@ -0,0 +1,29 @@
+#
+# Marvell OcteonTX2 drivers configuration
+#
+
+config OCTEONTX2_MBOX
+	tristate
+
+config OCTEONTX2_AF
+	tristate "Marvell OcteonTX2 RVU Admin Function driver"
+	select OCTEONTX2_MBOX
+	depends on ARM64 && PCI
+	---help---
+	  This driver supports Marvell's OcteonTX2 Resource Virtualization
+	  Unit's admin function manager which manages all RVU HW resources
+	  and provides a medium to other PF/VFs to configure HW. Should be
+	  enabled for other RVU device drivers to work.
+
+config OCTEONTX2_PF
+	tristate "Marvell OcteonTX2 NIC Physical Function driver"
+	select OCTEONTX2_MBOX
+	depends on PCI && ARM64 && ARM64_LSE_ATOMICS
+	---help---
+	  This driver supports Marvell's OcteonTX2 NIC physical function.
+
+config OCTEONTX2_VF
+	tristate "Marvell OcteonTX2 NIC Virtual Function driver"
+	depends on OCTEONTX2_PF
+	---help---
+	  This driver supports Marvell's OcteonTX2 NIC virtual function.
diff --git a/drivers/net/ethernet/marvell/octeontx2/Makefile b/drivers/net/ethernet/marvell/octeontx2/Makefile
index c855991c1a97..c54e25e6b9f9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/Makefile
+++ b/drivers/net/ethernet/marvell/octeontx2/Makefile
@@ -1,13 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 #
-# Makefile for Marvell's OcteonTX2 ethernet device drivers
+# Makefile for Marvell OcteonTX2 device drivers.
 #
 
-obj-$(CONFIG_OCTEONTX2_PF) += octeontx2_nicpf.o
-obj-$(CONFIG_OCTEONTX2_VF) += octeontx2_nicvf.o
-
-octeontx2_nicpf-y := otx2_pf.o otx2_common.o otx2_txrx.o otx2_ethtool.o \
-	otx2_ptp.o
-octeontx2_nicvf-y := otx2_vf.o
-
-ccflags-y += -I$(srctree)/drivers/soc/marvell/octeontx2
+obj-$(CONFIG_OCTEONTX2_AF) += af/
+obj-$(CONFIG_OCTEONTX2_PF) += nic/
diff --git a/drivers/soc/marvell/octeontx2/Makefile b/drivers/net/ethernet/marvell/octeontx2/af/Makefile
similarity index 100%
rename from drivers/soc/marvell/octeontx2/Makefile
rename to drivers/net/ethernet/marvell/octeontx2/af/Makefile
diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/cgx.c
rename to drivers/net/ethernet/marvell/octeontx2/af/cgx.c
diff --git a/drivers/soc/marvell/octeontx2/cgx.h b/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/cgx.h
rename to drivers/net/ethernet/marvell/octeontx2/af/cgx.h
diff --git a/drivers/soc/marvell/octeontx2/cgx_fw_if.h b/drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/cgx_fw_if.h
rename to drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h
diff --git a/drivers/soc/marvell/octeontx2/common.h b/drivers/net/ethernet/marvell/octeontx2/af/common.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/common.h
rename to drivers/net/ethernet/marvell/octeontx2/af/common.h
diff --git a/drivers/soc/marvell/octeontx2/mbox.c b/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/mbox.c
rename to drivers/net/ethernet/marvell/octeontx2/af/mbox.c
diff --git a/drivers/soc/marvell/octeontx2/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/mbox.h
rename to drivers/net/ethernet/marvell/octeontx2/af/mbox.h
diff --git a/drivers/soc/marvell/octeontx2/npc.h b/drivers/net/ethernet/marvell/octeontx2/af/npc.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/npc.h
rename to drivers/net/ethernet/marvell/octeontx2/af/npc.h
diff --git a/drivers/soc/marvell/octeontx2/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/npc_profile.h
rename to drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
diff --git a/drivers/soc/marvell/octeontx2/ptp.c b/drivers/net/ethernet/marvell/octeontx2/af/ptp.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/ptp.c
rename to drivers/net/ethernet/marvell/octeontx2/af/ptp.c
diff --git a/drivers/soc/marvell/octeontx2/ptp.h b/drivers/net/ethernet/marvell/octeontx2/af/ptp.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/ptp.h
rename to drivers/net/ethernet/marvell/octeontx2/af/ptp.h
diff --git a/drivers/soc/marvell/octeontx2/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu.c
diff --git a/drivers/soc/marvell/octeontx2/rvu.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu.h
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu.h
diff --git a/drivers/soc/marvell/octeontx2/rvu_cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_cgx.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_cpt.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_cpt.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_debugfs.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_ndc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_ndc.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_ndc.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_ndc.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_nix.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_npa.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npa.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_npa.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_npa.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_npc.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_ptp.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_ptp.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_ptp.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_ptp.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_reg.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_reg.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_reg.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_reg.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_reg.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu_reg.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_reg.h
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_reg.h
diff --git a/drivers/soc/marvell/octeontx2/rvu_sso.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_sso.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_struct.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu_struct.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_struct.h
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_struct.h
diff --git a/drivers/soc/marvell/octeontx2/rvu_tim.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_tim.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_validation.c
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
diff --git a/drivers/soc/marvell/octeontx2/rvu_validation.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.h
similarity index 100%
rename from drivers/soc/marvell/octeontx2/rvu_validation.h
rename to drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/Makefile b/drivers/net/ethernet/marvell/octeontx2/nic/Makefile
new file mode 100644
index 000000000000..a5819f513a81
--- /dev/null
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/Makefile
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: GPL-2.0
+#
+# Makefile for Marvell's OcteonTX2 ethernet device drivers
+#
+
+obj-$(CONFIG_OCTEONTX2_PF) += octeontx2_nicpf.o
+obj-$(CONFIG_OCTEONTX2_VF) += octeontx2_nicvf.o
+
+octeontx2_nicpf-y := otx2_pf.o otx2_common.o otx2_txrx.o otx2_ethtool.o \
+	otx2_ptp.o
+octeontx2_nicvf-y := otx2_vf.o
+
+ccflags-y += -I$(srctree)/drivers/net/ethernet/marvell/octeontx2/af
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_common.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.c
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_common.h
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_ptp.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_ptp.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_ptp.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.h
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_ptp.h
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_reg.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_reg.h
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_reg.h
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_reg.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_struct.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_struct.h
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_txrx.h
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
similarity index 100%
rename from drivers/net/ethernet/marvell/octeontx2/otx2_vf.c
rename to drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
diff --git a/drivers/soc/marvell/Kconfig b/drivers/soc/marvell/Kconfig
index 5886c1e040f2..25952bcee90f 100644
--- a/drivers/soc/marvell/Kconfig
+++ b/drivers/soc/marvell/Kconfig
@@ -4,17 +4,6 @@
 
 menu "Marvell SoC drivers"
 
-config OCTEONTX2_MBOX
-	tristate
-
-config OCTEONTX2_AF
-	tristate "OcteonTX2 RVU Admin Function driver"
-	select OCTEONTX2_MBOX
-	depends on ARM64 && PCI
-	---help---
-	  This driver supports Marvell's OcteonTX2 Resource Virtualization
-	  Unit's admin function manager which manages all RVU HW resources.
-
 config OCTEONTX2_RM
 	tristate "OcteonTX2 RVU Resource Manager driver"
 	depends on OCTEONTX2_AF
diff --git a/drivers/soc/marvell/Makefile b/drivers/soc/marvell/Makefile
index f8ab4dd1b8e7..9c9d47cbb084 100644
--- a/drivers/soc/marvell/Makefile
+++ b/drivers/soc/marvell/Makefile
@@ -1,3 +1,2 @@
 # SPDX-License-Identifier: GPL-2.0
-obj-y		+= octeontx2/
 obj-y		+= octeontx2-rm/
diff --git a/drivers/soc/marvell/octeontx2-rm/Makefile b/drivers/soc/marvell/octeontx2-rm/Makefile
index 6504ff39faef..2dc61fd4a8c8 100644
--- a/drivers/soc/marvell/octeontx2-rm/Makefile
+++ b/drivers/soc/marvell/octeontx2-rm/Makefile
@@ -7,4 +7,4 @@ obj-$(CONFIG_OCTEONTX2_RM) += octeontx2_rm.o
 
 octeontx2_rm-y := otx2_rm.o quota.o
 octeontx2_rm-$(CONFIG_OCTEONTX2_RM_DOM_SYSFS) += domain_sysfs.o
-ccflags-y += -I$(srctree)/drivers/soc/marvell/octeontx2
+ccflags-y += -I$(srctree)/drivers/net/ethernet/marvell/octeontx2/af
-- 
2.17.1

